### PR TITLE
Add native support for PlaylistItems

### PR DESCRIPTION
**Pull request by Phillip3**
Simulator Screen Shot - iPhone 11 - 2020-04-24 at 00 13 50

### CODE REVIEW

File: data/youtube.js [hunk 1]
```diff
@@ -10,53 +10,117 @@ const PLAYLIST_NAME_AVOID_LIST = [
...
+axios.interceptors.request.use(request => {
+  console.log('Starting Request', request)
+  return request
+})
+
+axios.interceptors.response.use(response => {
+  console.log('Response:', response)
+  return response
+})
+*/
```
Comment (dereckquock) [inline data/youtube.js:22]: you could guard this for dev env only 🤔 
```suggestion
if (__DEV__) {
  axios.interceptors.request.use(request => {
    console.log('Starting Request', request)
    return request
  })

  axios.interceptors.response.use(response => {
    console.log('Response:', response)
    return response
 })
}
```

File: navigation/AppNavigator.js [hunk 2]
```diff
@@ -72,6 +73,25 @@ const defaultOptions = {
   },
 };
 
+const MediaStack = createStackNavigator();
+
+function MediaStackScreen() {
```
Comment (dereckquock) [inline navigation/AppNavigator.js:85]: 🔥

File: screens/Media.js [hunk 3]
```diff
@@ -58,18 +59,78 @@ const MediaScreen = () => {
...
       setError(true);
       setLoading(false);
       Amplitude.logEventWithProperties('ERROR loading media', { error: err });
     }
   }
 
+  const takeToItem = (item) => {
+    const { id, title, description, thumbnails: { maxres = {} } = {} } = item;
+    console.log('about to take to Item');
+    console.log(item);
```
Comment (dereckquock) [inline screens/Media.js]:
```suggestion
```

File: screens/Media.js [hunk 4]
```diff
@@ -58,18 +59,78 @@ const MediaScreen = () => {
...
+    navigation.navigate('Playlist', {
+      playlistID: id,
+      playlistTitle: title,
+      playlistDescription: description,
+      playlistURI: maxres.url,
+    });
+  };
+
+  const PastSeriesSection = ({ sectionData = [] }) => {
+    if (sectionData === null || sectionData.length === 0) {
```
Comment (dereckquock) [inline screens/Media.js]:
```suggestion
    if (!sectionData || !sectionData.length) {
```

File: screens/Playlist.js [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+          MEDIA
+        </Text>
+        <View
+          style={[
+            styles.container,
+            { alignItems: 'center', justifyContent: 'center', padding: 16 },
+          ]}
+        >
+          <Title style={[{ textAlign: 'center' }]}>
+            {' '}
```
Comment (dereckquock) [inline screens/Playlist.js]:
```suggestion
```

File: screens/Playlist.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+            <Image
+              source={{ uri: playlistURI }}
+              style={styles.image}
+              resizeMode="cover"
+            />
+            <Text style={styles.description} numberOfLines={4}>
+              {data.length ? data[0].description : ''}
+            </Text>
+            <Subtitle style={styles.subtitle}>IN THIS SERIES</Subtitle>
+            {/* {
```
Comment (dereckquock) [inline screens/Playlist.js]: 💁  **suggstion**
if we don't need this, we can probably just delete it. we can always see git history if we need it back again.

File: screens/Playlist.js [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+  },
+  item: {
+    width: screenWidth,
+    flexDirection: 'row',
+  },
+  title: {
+    paddingTop: 0,
+    paddingLeft: 16,
+    paddingRight: 16,
+    // fontSize: 16,
```
Comment (dereckquock) [inline screens/Playlist.js]:
```suggestion
```

**Review by dereckquock — CHANGES_REQUESTED** [review]
🔥 good stuff!!

File: data/youtube.js (hunk 1 above)
Comment (Phillip3) [reply data/youtube.js:22]: @dereckquock I don't want this to run in dev all the time--it fills up the output, but it's really convenient to have if you're debugging something related to fetching.

File: screens/Playlist.js [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+                      {title}
+                    </Heading>
+                    <Text style={[styles.description, {paddingTop: 8}]}>
+                    {new Date(publishDate).toLocaleDateString()}
+                    </Text>
+                    <View style={{flex:1}}/>
+                  </View>
+                </View>
+              </TouchableHighlight>
+            ))
```
Comment (Phillip3) [inline screens/Playlist.js]: @dereckquock I don't know why this map replacement for the flatlist isn't working.

**Review by dereckquock — APPROVED** [review]

**Pull request merged**